requirements.txt

```diff
@@ -1,6 +1,7 @@
 numpy
 pyserial
 pyvisa>=1.9
+pyvisa-py
```

Comment:
Ah interesting, `pyvisa` supports multiple backends. Is `pyvisa-py` not the default? What does specifying this in the `requirements.txt` solve?

Reply:
The oscilloscope requires the NI VISA style, thus `pyvisa-py`. With the current `requirements.txt` I received an error to use `pyvisa-py` which was not installed (after running `pip install -r requirements.txt`). After also installing `pyvisa-py`, everything worked out of the box. Thus I figured I'll add it to the requirements. 
Interesting: the `pyvisa-py` [docs](https://pyvisa-py.readthedocs.io/en/latest/) state that one needs to add `'@py'` when calling the resource manager. However, I never had to specifically do that, installing it was enough.

Reply:
Yeah I saw that in the docs, so I'm confused :/

I do wonder if it will then impact anyone using the current VISA impl. If we do want to include it, we should add it to `setup.py` and add some minimum versioning to both references.

Reply:
I agree, it is confusing to me as well. It could be added as extra requirements or simply noted at the beginning of the docstring. I'm also a bit worried that adding it by default changes the default behavior of VISA. I don't have a good way of testing it at the moment with hardware... Need to do some more reading on this one.

Reply:
From the [PyVisa documentation](https://pyvisa.readthedocs.io/en/latest/introduction/configuring.html) on configuring the backend:

 > Currently there are two backends available: The one included in pyvisa, which uses the IVI library (include NI-VISA, Keysight VISA, R&S VISA, tekVISA etc.), and the backend provided by pyvisa-py, which is a pure python implementation of the VISA library. If no backend is specified, pyvisa uses the IVI backend if any IVI library has been installed (see next section for details). Failing that, it uses the pyvisa-py backend.

As I understand it: PyVisa uses the standard VISA interface (IVI), unless it fails, then pyvisa-py kicks in. That would explain why I don't have to specify it. The same page states just further down in the note:

 > The IVI backend requires that you install first the IVI-VISA library. For example you can use NI-VISA or any other library in your opinion. [...]

I did not configure that on my system, that might be why I had to install `pyvisa-py`. Reading this, let me test how / if this works with NI-VISA. Don't have the scope handy till next week, but I can test it easily then. I think it would make most sense to remove `pyvisa-py` from the `requirements.txt` and write a more detailed introduction in class intro docstring, where the communication protocol is discussed as well. If NI-VISA works (which is what I suspect), `pyvisa-py` is strictly speaking not a requirement but an option and the user should ensure that a working VISA library is installed.

Reply:
I agree with your comments. Maybe `pyvisa-py` can be a extra dep in `setup.py`. But that can be explored separately from this PR.

Reply:
This might fall under the same category as mentioned in #91. Agree, should be explored somewhere else.
I'll change `requirements.txt` and the docstring pointing to the VISA library discussion after testing with the hardware next week and update the PR.

Reply:
Yes, pyvisa-py is the fallback if pyvisa does not find another (e.g. NI) library installed. If then pyvisa also does not find pyvisa-py, it prompts you to install that (i.e. that message should have originated from pyvisa). Might make most sense to just point to the relevant pyvisa docs.